src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1315,20 +1315,25 @@ CHIP_ERROR CASESession::GetHardcodedTime()
 
 CHIP_ERROR CASESession::SetEffectiveTime()
 {
-    System::Clock::Milliseconds64 currentTimeMS;
-    CHIP_ERROR err = System::SystemClock().GetClock_RealTimeMS(currentTimeMS);
-    if (err == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+    System::Clock::Milliseconds64 timeMS;
+    CHIP_ERROR err = System::SystemClock().GetClock_RealTimeMS(timeMS);
+    if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(
-            SecureChannel,
-            "The device does not support GetClock_RealTimeMS() API. This will eventually result in CASE session setup failures.");
-        // TODO: Remove use of hardcoded time during CASE setup
-        return GetHardcodedTime();
+        ChipLogDetail(SecureChannel, "The device does not support GetClock_RealTimeMS() API. Trying to use last known good time.");
+
+        err = System::SystemClock().GetClock_LastKnownGoodTimeMS(timeMS);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(SecureChannel,
+                         "The device does not support GetClock_RealTimeMS() API and the last known good time is not set. This will "
+                         "eventually result in CASE session setup failures.");
```

Comment:
This should just yield the change of the validation policy to ignore expiry. Otherwise, you can get into a corner where it's impossible to establish CASE since last known good time is too early or somehow unset, due to the dependency chain path I mentioned above regarding the location of the time setting.

Reply:
I agree and this is a blocking issue for our usage of the SDK.  I will create a new issue for capturing further details and corresponding evaluation of the spec on certificate expiration.